### PR TITLE
Xfce 4.18

### PR DESCRIPTION
**Pull request by muscaln**
###### Description of changes

Update xfce

- Closes #206423

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by PedroHLC** [comment]
Result of `nixpkgs-review pr 206282` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>5 packages marked as broken and skipped:</summary>
  <ul>
    <li>xfce.thunar-bare</li>
    <li>xfce.xfce4-embed-plugin</li>
    <li>xfce.xfce4-hardware-monitor-plugin</li>
    <li>xfce.xfce4-namebar-plugin</li>
    <li>xmonad_log_applet_xfce</li>
  </ul>
</details>
<details>
  <summary>4 packages failed to build:</summary>
  <ul>
    <li>xfce.xfce4-cpufreq-plugin</li>
    <li>xfce.xfce4-cpugraph-plugin</li>
    <li>xfce.xfce4-sensors-plugin</li>
    <li>xfce.xfce4-volumed-pulse</li>
  </ul>
</details>
<details>
  <summary>58 packages built:</summary>
  <ul>
    <li>caffeine-ng</li>
    <li>lightdm-gtk-greeter</li>
    <li>pragha</li>
    <li>xfce.catfish</li>
    <li>xfce.exo</li>
    <li>xfce.garcon</li>
    <li>xfce.gigolo</li>
    <li>xfce.libxfce4ui</li>
    <li>xfce.libxfce4util</li>
    <li>xfce.mousepad</li>
    <li>xfce.orage</li>
    <li>xfce.parole</li>
    <li>xfce.ristretto</li>
    <li>xfce.thunar</li>
    <li>xfce.thunar-archive-plugin</li>
    <li>xfce.thunar-dropbox-plugin</li>
    <li>xfce.thunar-media-tags-plugin</li>
    <li>xfce.thunar-volman</li>
    <li>xfce.tumbler</li>
    <li>xfce.xfburn</li>
    <li>xfce.xfce4-appfinder</li>
    <li>xfce.xfce4-battery-plugin</li>
    <li>xfce.xfce4-clipman-plugin</li>
    <li>xfce.xfce4-datetime-plugin</li>
    <li>xfce.xfce4-dev-tools</li>
    <li>xfce.xfce4-dict</li>
    <li>xfce.xfce4-dockbarx-plugin</li>
    <li>xfce.xfce4-eyes-plugin</li>
    <li>xfce.xfce4-fsguard-plugin</li>
    <li>xfce.xfce4-genmon-plugin</li>
    <li>xfce.xfce4-i3-workspaces-plugin</li>
    <li>xfce.xfce4-mailwatch-plugin</li>
    <li>xfce.xfce4-mpc-plugin</li>
    <li>xfce.xfce4-netload-plugin</li>
    <li>xfce.xfce4-notes-plugin</li>
    <li>xfce.xfce4-notifyd</li>
    <li>xfce.xfce4-panel</li>
    <li>xfce.xfce4-panel-profiles</li>
    <li>xfce.xfce4-power-manager</li>
    <li>xfce.xfce4-pulseaudio-plugin</li>
    <li>xfce.xfce4-screensaver</li>
    <li>xfce.xfce4-screenshooter</li>
    <li>xfce.xfce4-session</li>
    <li>xfce.xfce4-settings</li>
    <li>xfce.xfce4-systemload-plugin</li>
    <li>xfce.xfce4-taskmanager</li>
    <li>xfce.xfce4-terminal</li>
    <li>xfce.xfce4-time-out-plugin</li>
    <li>xfce.xfce4-timer-plugin</li>
    <li>xfce.xfce4-verve-plugin</li>
    <li>xfce.xfce4-weather-plugin</li>
    <li>xfce.xfce4-whiskermenu-plugin</li>
    <li>xfce.xfce4-windowck-plugin</li>
    <li>xfce.xfce4-xkb-plugin</li>
    <li>xfce.xfconf</li>
    <li>xfce.xfdashboard</li>
    <li>xfce.xfdesktop</li>
    <li>xfce.xfwm4</li>
  </ul>
</details>

**Comment by PedroHLC** [comment]
Result of `nixpkgs-review pr 206282` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>7 packages marked as broken and skipped:</summary>
  <ul>
    <li>catfish</li>
    <li>lightdm_gtk_greeter</li>
    <li>xfce.thunar-bare</li>
    <li>xfce.xfce4-embed-plugin</li>
    <li>xfce.xfce4-hardware-monitor-plugin</li>
    <li>xfce.xfce4-namebar-plugin</li>
    <li>xmonad_log_applet_xfce</li>
  </ul>
</details>
<details>
  <summary>62 packages built:</summary>
  <ul>
    <li>caffeine-ng</li>
    <li>lightdm-gtk-greeter</li>
    <li>pragha</li>
    <li>xfce.catfish</li>
    <li>xfce.exo</li>
    <li>xfce.garcon</li>
    <li>xfce.gigolo</li>
    <li>xfce.libxfce4ui</li>
    <li>xfce.libxfce4util</li>
    <li>xfce.mousepad</li>
    <li>xfce.orage</li>
    <li>xfce.parole</li>
    <li>xfce.ristretto</li>
    <li>xfce.thunar</li>
    <li>xfce.thunar-archive-plugin</li>
    <li>xfce.thunar-dropbox-plugin</li>
    <li>xfce.thunar-media-tags-plugin</li>
    <li>xfce.thunar-volman</li>
    <li>xfce.tumbler</li>
    <li>xfce.xfburn</li>
    <li>xfce.xfce4-appfinder</li>
    <li>xfce.xfce4-battery-plugin</li>
    <li>xfce.xfce4-clipman-plugin</li>
    <li>xfce.xfce4-cpufreq-plugin</li>
    <li>xfce.xfce4-cpugraph-plugin</li>
    <li>xfce.xfce4-datetime-plugin</li>
    <li>xfce.xfce4-dev-tools</li>
    <li>xfce.xfce4-dict</li>
    <li>xfce.xfce4-dockbarx-plugin</li>
    <li>xfce.xfce4-eyes-plugin</li>
    <li>xfce.xfce4-fsguard-plugin</li>
    <li>xfce.xfce4-genmon-plugin</li>
    <li>xfce.xfce4-i3-workspaces-plugin</li>
    <li>xfce.xfce4-mailwatch-plugin</li>
    <li>xfce.xfce4-mpc-plugin</li>
    <li>xfce.xfce4-netload-plugin</li>
    <li>xfce.xfce4-notes-plugin</li>
    <li>xfce.xfce4-notifyd</li>
    <li>xfce.xfce4-panel</li>
    <li>xfce.xfce4-panel-profiles</li>
    <li>xfce.xfce4-power-manager</li>
    <li>xfce.xfce4-pulseaudio-plugin</li>
    <li>xfce.xfce4-screensaver</li>
    <li>xfce.xfce4-screenshooter</li>
    <li>xfce.xfce4-sensors-plugin</li>
    <li>xfce.xfce4-session</li>
    <li>xfce.xfce4-settings</li>
    <li>xfce.xfce4-systemload-plugin</li>
    <li>xfce.xfce4-taskmanager</li>
    <li>xfce.xfce4-terminal</li>
    <li>xfce.xfce4-time-out-plugin</li>
    <li>xfce.xfce4-timer-plugin</li>
    <li>xfce.xfce4-verve-plugin</li>
    <li>xfce.xfce4-volumed-pulse</li>
    <li>xfce.xfce4-weather-plugin</li>
    <li>xfce.xfce4-whiskermenu-plugin</li>
    <li>xfce.xfce4-windowck-plugin</li>
    <li>xfce.xfce4-xkb-plugin</li>
    <li>xfce.xfconf</li>
    <li>xfce.xfdashboard</li>
    <li>xfce.xfdesktop</li>
    <li>xfce.xfwm4</li>
  </ul>
</details>

**Review by PedroHLC — APPROVED** [review]

**Review by romildo — APPROVED** [review]
Looks good to me. Using on my notebook now.

File: pkgs/desktops/xfce/core/thunar/default.nix [hunk 1]
```diff
@@ -44,9 +44,9 @@ let unwrapped = mkXfceDerivation {
     xfconf
   ];
 
-  patches = [
-    ./thunarx_plugins_directory.patch
```
Comment (bobby285271) [inline pkgs/desktops/xfce/core/thunar/default.nix:48]: Can we remove this [patch](https://github.com/NixOS/nixpkgs/blob/d46fa9eb57272b1eb58646ea5075c45203a18c7f/pkgs/desktops/xfce/core/thunar/thunarx_plugins_directory.patch) now?

File: pkgs/desktops/xfce/core/thunar/default.nix [hunk 2]
```diff
@@ -44,9 +44,9 @@ let unwrapped = mkXfceDerivation {
...
-  ];
+  configureFlags = [ "--with-custom-thunarx-dirs-enabled" ];
+
+  NIX_CFLAGS_COMPILE = "-I${libxfce4ui.dev}/include/xfce4";
```
Comment (bobby285271) [inline pkgs/desktops/xfce/core/thunar/default.nix:49]: Not a change request: *I am quite curious in why this is now needed and I just filed https://gitlab.xfce.org/xfce/thunar/-/issues/972 (haven't filed an issue for `xfce4-terminal` yet but I think they are the same cause)* :thinking:

**Comment by bobby285271** [comment]
@ofborg test xfce

File: pkgs/desktops/xfce/core/thunar/default.nix (hunk 1 above)
Comment (muscaln) [reply pkgs/desktops/xfce/core/thunar/default.nix:48]: I thought i deleted it, Thanks!

**Review by romildo — APPROVED** [review]

**Pull request merged**